.github/workflows/windows.yml

```diff
@@ -0,0 +1,93 @@
+name: mingw-w64
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  mingw-w64:
+    name: Build and test
+    runs-on: windows-latest
+    steps:
+      - name: Checkout Git branch
+        uses: actions/checkout@v2
+
+      - name: Cache Msys2
+        id: cache-msys2
+        uses: actions/cache@v1.1.0
+        with:
+          path: msys64
+          key: ${{ runner.os }}-msys64
+
+      - name: Get Msys2
+        if: steps.cache-msys2.outputs.cache-hit != 'true'
+        run: |
+          Invoke-WebRequest -Outfile msys2.tar.xz -Uri "https://github.com/msys2/msys2-installer/releases/download/nightly-x86_64/msys2-base-x86_64-latest.tar.xz"
+          xz -d msys2.tar.xz
+          tar -xf msys2.tar
+          rm msys2.tar
+
+      - name: Add Msys2 to Path
+        run: |
+          echo "::add-path::${env:GITHUB_WORKSPACE}/msys64/mingw64/bin"
+          echo "::add-path::${env:GITHUB_WORKSPACE}/msys64/usr/bin"
+
+      - shell: cmd
+        if: steps.cache-msys2.outputs.cache-hit != 'true'
+        run: bash -lc 'exit'
+
+      - name: Update Pacman
+        run: pacman -Syy --needed --ask=20 --noconfirm
```

Comment:
I just left `--ask=20` in there since I split this command off the other pacman command to update the packages themselves to see if that would influence the caching job right below it. I'm not aware of any instances where `-Syy` would fail or ask for confirmation, so I think I can just remove it